### ISSUE TITLE
Parallel build fails with `Rengine.o: file not recognized: File truncated`

### ISSUE THREAD

**Issue by paulmenzel** [s-u/rJava#182]
Building R 3.5.3 and trying to install rJava it fails with the error below.

     export MAKEFLAGS="-j $(nproc)"

```
[…]
trying URL 'https://cran.rstudio.com/src/contrib/rJava_0.9-11.tar.gz'
Content type 'application/x-gzip' length 675188 bytes (659 KB)
==================================================
downloaded 659 KB
[…]
* installing *source* package ‘rJava’ ...
** package ‘rJava’ successfully unpacked and MD5 sums checked
[…]
if [ -n "/usr/local/bin/javah" ]; then /usr/local/bin/javah -d . -classpath . org.rosuda.JRI.Rengine; fi
if [ -n "/usr/local/bin/javah" ]; then /usr/local/bin/javah -d . -classpath . org.rosuda.JRI.Rengine; fi
gcc -c -o Rengine.o Rengine.c -g -Iinclude  -DRIF_HAS_CSTACK -DRIF_HAS_RSIGHAND -g -O2  -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux  -fpic -I/usr/local/java/jre/.
./include -I/usr/local/java/jre/../include/linux -I/pkg/R-3.5.3-0/lib64/R/include -I/pkg/R-3.5.3-0/lib64/R/include
gcc -c -o Rcallbacks.o Rcallbacks.c -g -Iinclude  -DRIF_HAS_CSTACK -DRIF_HAS_RSIGHAND -g -O2  -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux  -fpic -I/usr/local/
java/jre/../include -I/usr/local/java/jre/../include/linux -I/pkg/R-3.5.3-0/lib64/R/include -I/pkg/R-3.5.3-0/lib64/R/include
gcc -c -o Rengine.o Rengine.c -g -Iinclude  -DRIF_HAS_CSTACK -DRIF_HAS_RSIGHAND -g -O2  -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux  -fpic -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux -I/pkg/R-3.5.3-0/lib64/R/include -I/pkg/R-3.5.3-0/lib64/R/include
gcc -c -o Rcallbacks.o Rcallbacks.c -g -Iinclude  -DRIF_HAS_CSTACK -DRIF_HAS_RSIGHAND -g -O2  -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux  -fpic -I/usr/local/java/jre/../include -I/usr/local/java/jre/../include/linux -I/pkg/R-3.5.3-0/lib64/R/include -I/pkg/R-3.5.3-0/lib64/R/include
gcc -o libjri.so Rengine.o jri.o Rcallbacks.o Rinit.o globals.o rjava.o  -shared -L/usr/local/java/jre/lib/amd64/server -ljvm -Wl,--export-dynamic -fopenmp -Wl,-rpath,/pkg/R-3.5.3-0/lib64/R/lib -L/pkg/R-3.5.3-0/lib64/R/lib -lR -lpcre -llzma -lbz2 -lz -lrt -ldl -lm -licuuc -licui18n
Rengine.o: file not recognized: File truncated
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile.all:35: libjri.so] Error 1
make[2]: Leaving directory '/dev/shm/R-3.5.3-0.build.tmp/Rtmpyy5tBp/R.INSTALL20cd78c939a3/rJava/jri/src'
make[1]: *** [Makefile.all:22: src/libjri.so] Error 2
make[1]: *** Waiting for unfinished jobs....
gcc -o libjri.so Rengine.o jri.o Rcallbacks.o Rinit.o globals.o rjava.o  -shared -L/usr/local/java/jre/lib/amd64/server -ljvm -Wl,--export-dynamic -fopenmp -Wl,-rpath,/pkg/R-3.5.3-0/lib64/R/lib -L/pkg/R-3.5.3-0/lib64/R/lib -lR -lpcre -llzma -lbz2 -lz -lrt -ldl -lm -licuuc -licui18n
/usr/local/bin/javac -target 1.4 -source 1.4 -d . ../RMainLoopCallbacks.java ../Rengine.java ../RConsoleOutputStream.java ../RVector.java ../RBool.java ../Mutex.java ../RList.java ../REXP.java ../RFactor.java ../package-info.java
warning: [options] bootstrap class path not set in conjunction with -source 1.4
warning: [options] source value 1.4 is obsolete and will be removed in a future release
warning: [options] target value 1.4 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
4 warnings
/usr/local/bin/jar fc JRI.jar org libjri.so
make[2]: Leaving directory '/dev/shm/R-3.5.3-0.build.tmp/Rtmpyy5tBp/R.INSTALL20cd78c939a3/rJava/jri/src'
make[1]: Leaving directory '/dev/shm/R-3.5.3-0.build.tmp/Rtmpyy5tBp/R.INSTALL20cd78c939a3/rJava/jri'
make: *** [Makevars:14: jri] Error 2
ERROR: compilation failed for package ‘rJava’
* removing ‘/pkg/R-3.5.3-0/lib64/R/library/rJava’
```

**Comment by paulmenzel** [comment]
The Make target prerequisites look fine though.

https://github.com/s-u/rJava/blob/53accf829a81383acc142a55ba4d462e321aca06/jri/src/Makefile.all#L34-L35

**Comment by paulmenzel** [comment]
Thank you for fixing (working around) the problem.

For R 3.5.3 I get:

    package ‘rJava’ is not available (for R version 3.5.3)

But with R 3.6 it work:

```
$ R

R version 3.6.3 (2020-02-29) -- "Holding the Windsock"
[…]
> install.packages("rJava")
[…]
Selection: 41
trying URL 'https://ftp.gwdg.de/pub/misc/cran/src/contrib/rJava_1.0-6.tar.gz'
Content type 'application/octet-stream' length 1047062 bytes (1022 KB)
==================================================
downloaded 1022 KB

* installing *source* package ‘rJava’ ...
** package ‘rJava’ successfully unpacked and MD5 sums checked
** using staged installation
checking for gcc... gcc
[…]
make[1]: Leaving directory '/scratch/local/RtmpexWGuG/R.INSTALL421829393b4f/rJava/jri'
installing via 'install.libs.R' to /home/pmenzel/R/x86_64-pc-linux-gnu-library/3.6/00LOCK-rJava/00new/rJava
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (rJava)
[…]
```

**Comment by paulmenzel** [comment]
Though it’s strange, as the fix is not in [rJava 1.0-6, released in December 2021](https://rforge.net/rJava/news.html).
